### DISCUSSION TITLE
API next / previous item

### DISCUSSION THREAD

**Discussion by ernst86913** [directus/directus#8265] [Archive]
Hi -
Implemented directus api integration on my web site.

When browsing a post of my blog, I’d like to be able to provide previous / next button (based on query in my paginated posts overview page)

1) is there a way to get the index of an item in a specific query to be able to derive the index of previous / next item?

2) is there a way to query limit=3 with current item in the middle?

3) any other ideas / approaches?

thanks a lot in advance,
Roman

**Comment by joselcvarela** [comment]
Hello,
1. The index of the item is dependable on your query, so it is not possible
2. No it is not possible

What you can do is to do the query and add `limit=1`.
Then you should save the current page number and when user clicks on next you add `page=${currentPage+1}` or if he clicks on previous button `page=${currentPage -1}`

**Reply by benhaynes** [reply]
A few relevant Discussions:

https://github.com/directus/directus/discussions/6609
https://github.com/directus/directus/discussions/5348

This pagination metadata could also have "next" and "previous" links...
